src/Launcher.vala

```diff
@@ -336,6 +342,43 @@ public class Dock.Launcher : Gtk.Box {
         }
     }
 
+    private void close_all_instances(){
+        if (app.running) {
+            if (app.action_group.lookup_action(CLOSE_WINDOWS_ACTION) != null) {
+                return;
+            }
+            
+            var close_item = new GLib.MenuItem(_(CLOSE_WINDOWS_LABEL), ACTION_PREFIX + CLOSE_WINDOWS_ACTION);
+            app.menu_model.append_item(close_item);
+    
+            var close_action = new SimpleAction(CLOSE_WINDOWS_ACTION, null);
+            close_action.activate.connect(() => {
+                var desktop_integration = LauncherManager.get_default().desktop_integration;
+            
+                foreach (var win in app.windows) {
+                    desktop_integration.close_window.begin(win.uid, null);
+                }
+        
+                app.windows.clear(); 
```

Comment:
```suggestion

```

The list will be automatically updated by gala if windows close. We shouldn't do it manually here because sometimes windows won't close on request but present a dialog asking for some confirmation (e.g. if a process is running in terminal)